README.md

```diff
@@ -77,6 +83,9 @@ Seq.of(1, 2, 3).join("|", "^", "$");
 // (tuple(1, 1), tuple(2, 2), tuple(4, null))
 Seq.of(1, 2, 4).leftOuterJoin(Seq.of(1, 2, 3), (a, b) -> a == b);
 
+// (tuple(tuple(1, 0), NULL), tuple(tuple(2, 1), tuple(1, 0)))
+Seq.of(new Tuple2<Integer, Integer>(1, 0), new Tuple2<Integer, Integer>(2, 1)).leftOuterSelfJoin((t, u) -> t.v2 == u.v1)

```

Comment:
For brevity's sake, we can replace the constructor call with the `Tuple.tuple()` convenience method and assume a static import, so:

```java
Seq.of(tuple(1, 0), tuple(2, 1))...
```